API-9080 Add require_every_value to list_archives and reports filters (#983)
livechat/livechat-public-docs: content/data-reporting/reports-api/v3.4/index.mdx

List of default properties can be found [**here**](/messaging/references/default-properties/#reference).

**2)** `<string_filter_type>` can take following values:

- `exists` (`bool`)
- `values` (`string[]` - an array of strings)
- `exclude_values` (`string[]` - an array of strings)
- `require_every_value` (`bool`) - if set to `true` will return chats that have all elements of `values` or `exclude_values` in the field that the filter applies to

Only one of `exists`, `values` or `exclude_values` may be passed.

**3)** `<integer_filter_type>` can take following values:

- `exists` (`bool`)
- `values` (`int[]` - an array of integers)
- `exclude_values` (`int[]` - an array of integers)

> - `require_every_value` (`bool`) - if set to `true` will return chats that have all elements of `values` or `exclude_values` in the field that the filter applies to

Reviewer: ```suggestion
- `require_every_value` (`bool`) - if set to `true`, will return only those chats that have all elements passed in `values` or `exclude_values`
```

> Only one of `exists`, `values` or `exclude_values` may be passed.

Reviewer: ```suggestion
You can pass only one of the following values at a time: `exists`, `values` or `exclude_values`.
```

**4)** `<survey>` contains the following fields:

- `type` (`string`) - allowed values: `pre_chat`, `post_chat`
- `answer_id` (`string`)

**5)** `<event_type_filter_type>` can take following values:

- `values` (`string[]` - an array of [Event](/messaging/agent-chat-api/v3.4/#events) types, duplicates are ignored)
- `exclude_values` (`string[]` - an array of [Event](/messaging/agent-chat-api/v3.4/#events) types, duplicates are ignored)

> - `require_every_value` (`bool`) - if set to `true` will return chats that have all elements of `values` or `exclude_values` in the field that the filter applies to

Reviewer: ```suggestion
- `require_every_value` (`bool`) - if set to `true`, will return only those chats that have all elements passed in `values` or `exclude_values`
```

> Only one of `exists`, `values` or `exclude_values` may be passed.

Reviewer: ```suggestion
You can pass only one of the following values at a time: `exists`, `values` or `exclude_values`.
```
